components/proxy/src/test/java/com/hotels/styx/StyxServerTest.java

```diff
@@ -110,15 +111,14 @@ public void disablesResourceLeakDetectionByDefault() {
     }
 
     @Test
-    public void stopsTheServerWhenPluginFailsToStart() {
+    public void stopsTheServerWhenPluginFailsToStart() throws TimeoutException {
         StyxServer styxServer = null;
         try {
             styxServer = styxServerWithPlugins(ImmutableMap.of(
                     "foo", new NonStarterPlugin("foo"),
                     "mockplugin3", mock(Plugin.class)));
 
-            Service service = styxServer.startAsync();
-            eventually(() -> assertThat(service.state(), is(FAILED)));
+            tryToStartExpectingFailure(styxServer);
```

Comment:
This change seems unnecessary? 

Ok, this is a matter of taste, but I like the old style better where the test body explicitly specifies the desired behaviour. Without hiding it into `tryToStartExpectingFailure` method. Especially given such a marginal reduction in lines of code.

Reply:
The test was changed because `StyxServer.startAsync` is actually asynchronous now. In the past it was actually blocking.

The new `tryToStartExpectingFailure` looks like:
```
        try {
            server.startAsync().awaitRunning(3, SECONDS);
            fail("Expected failure but succeeded");
        } catch (IllegalStateException e) {
            // Desired behaviour, so no action taken
        }
```
So reducing that to only 1 line is quite substantial.